packages/framework/collabspaces/src/collabSpaces.ts

```diff
@@ -415,9 +417,35 @@ export class TempCollabSpaceRuntime
 		}
 		this.matrix.switchSetCellPolicy();
 
+		// We need to ensure that we are tracking all the cells and when removing cells, we need to
+		// remove the tracking information as well as update the indexes in the map.
+		const removeCellsFromMap = (start: number, count: number, map: Map<string, any>) => {
+			if (count > 0) {
+				// Convert the map to an array of key-value pairs
+				const entries = Array.from(map.entries());
+
+				// Remove the specified range of items from the array
+				// Note the index from the reverse mapping is off by 1 as the matrix has a row and col tracking IDs
+				// and the map indexes are 0 based
+				entries.splice(start - 2, count);
+
+				// Clear the original map
+				map.clear();
+
+				// Re-populate the map with the remaining items from the array
+				entries.forEach(([key, value], index) => {
+					map.set(key, index + 1);
+				});
+			}
+		};
+
 		this.matrix.openMatrix({
-			rowsChanged: (rowStart: number, removedCount: number, insertedCount: number) => {},
-			colsChanged: (colStart: number, removedCount: number, insertedCount: number) => {},
+			rowsChanged: (rowStart: number, removedCount: number, insertedCount: number) => {
+				removeCellsFromMap(rowStart, removedCount, this.rowMap);
```

Comment:
We get here for both cases - rows added, as well as rows deleted. I think the call as it stands only deals with removals.

Reply:
rows added do not help as we cant add blank rows

Reply:
Or is logic in insertCellIntoReverseMap() tries to enforce some order? If it does, are you sure that map.entries() provides us some guarantees? I do not see anything listed on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/entries, though I have not looked at full spec

Reply:
Oh, I see, the issue with inserts - we expect to get IDs from remote clients. Maybe at least add some comments on that, as it's a bit hard to follow.